sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryClient.java

```diff
@@ -406,32 +406,70 @@ public DirectorySetMetadataInfo setMetadata(Map<String, String> metadata) {
+     * @param handleId Handle ID to be closed.
+     */
+    public void forceCloseHandle(String handleId) {
+        forceCloseHandleWithResponse(handleId, null, Context.NONE);
+    }
+
+    /**
+     * Closes a handle on the directory at the service. This is intended to be used alongside {@link
+     * #listHandles(Integer, boolean, Duration, Context)}.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Force close handles returned by list handles.</p>
+     *
+     * {@codesnippet com.azure.storage.file.DirectoryClient.forceCloseHandleWithResponse#String-Duration-Context}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/force-close-handles">Azure Docs</a>.</p>
+     *
+     * @param handleId Handle ID to be clsoed.
      * @param timeout An optional timeout applied to the operation. If a response is not returned before the timeout
      * concludes a {@link RuntimeException} will be thrown.
      * @param context Additional context that is passed through the Http pipeline during the service call.
-     * @return The counts of number of handles closed.
-     * @throws RuntimeException if the operation doesn't complete before the timeout concludes.
+     * @return A response that only contains headers and response status code.
      */
-    public PagedIterable<Integer> forceCloseHandles(String handleId, boolean recursive, Duration timeout,
-        Context context) {
-        return new PagedIterable<>(directoryAsyncClient
-            .forceCloseHandlesWithOptionalTimeout(handleId, recursive, timeout, context));
+    public Response<Void> forceCloseHandleWithResponse(String handleId, Duration timeout, Context context) {
+        return Utility.blockWithOptionalTimeout(directoryAsyncClient
+            .forceCloseHandleWithResponse(handleId, context), timeout);
+    }
+
+    /**
+     * Closes all handles opened on the directory at the service.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Force close all handles recursively.</p>
+     *
+     * {@codesnippet com.azure.storage.file.DirectoryClient.forceCloseAllHandles#boolean-Duration-Context}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/force-close-handles">Azure Docs</a>.</p>
+     *
+     * @param recursive Flag indicating if the operation should apply to all subdirectories and files contained in the
+     * directory.
+     * @param timeout An optional timeout applied to the operation. If a response is not returned before the timeout
+     * concludes a {@link RuntimeException} will be thrown.
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return The number of handles closed.
+     */
+    public int forceCloseAllHandles(boolean recursive, Duration timeout, Context context) {
```

Comment:
No `withResponse` variant?

Reply:
Direction for this API is to just return the number of handles closed as we are abstracting away the pagedness of the REST API.